mitodl/mitxpro
# Trigger hubspot celery tasks where appropriate

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #314

#### What's this PR do?
- Triggers a hubspot sync task when a `User` is created/updated via `UserSerializer` or the registration pipeline, if `HUBSPOT_API_KEY` is set
- Triggers  a hubspot sync task when a `Line`, or `ProductVersion` is created/updated via overridden `save` method, , if `HUBSPOT_API_KEY` is set
- Triggers a hubspot sync task when appropriate for `Order` (`create_unfulfilled_order` API function, checkout and order fulfillment views).

#### How should this be manually tested?
Tests should pass; you should be able to create/update users, orders, lines, and products/product versions without anything breaking, with or without `HUBSPOT_API_KEY` set.

#### Any background context you want to provide?
- Decided to trigger `User` syncs in serializers/pipeline instead of overriding model saves, because there are 3 models (`User`, `LegalAddress`, `Profile`) involved and there should only be 1 sync if all 3 are changed at once.
- Decided to sync `Product` on update/creation of `ProductVersion`, which contains most of the relevant information (price, description) that is most likely to change.
